spec/features/company_user_signup_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.feature "CompanyUserSignUp", type: :feature do
+  describe("Company signup creates a user and a company") do
+    let! (:state) { State.create(abbr: "CO", id: 14) }
+
+    let! (:usa_city1) { UsaCity.create(name: "Denver", state_id: 14) }
+
+    it "can signup" do
+      allow(User).to receive(:generate_password).and_return("abc123abc")
```

Comment:
Yes, this is stubbing the User class method called generate_password.  Since the password is never stored as plain text in our database, there's no way for us to know what the password is when a user goes through the signup then login process.  We had to stub the method that generates passwords in order to test that the signup and login process works.